core/src/main/scala/kafka/controller/ControllerChannelManager.scala

```diff
@@ -395,9 +398,9 @@ class ControllerBrokerRequestBatch(controller: KafkaController) extends  Logging
               val endPoints = broker.endPoints.map { case (securityProtocol, endPoint) =>
                 securityProtocol -> new UpdateMetadataRequest.EndPoint(endPoint.host, endPoint.port)
               }
-              new UpdateMetadataRequest.Broker(broker.id, endPoints.asJava)
+              new UpdateMetadataRequest.Broker(broker.id, endPoints.asJava, broker.rack.orNull)
             }
-            new UpdateMetadataRequest(controllerId, controllerEpoch, partitionStates.asJava, liveBrokers.asJava)
+            new UpdateMetadataRequest(version, controllerId, controllerEpoch, partitionStates.asJava, liveBrokers.asJava)
```

Comment:
@junrao I think we can use the same constructor for both v1 and v2 except for the version number. When the request is serialized, the rack in v1 request is ignored according to Protocol. In other words, regardless v1 or v2, we can have rack in UpdateMetadataRequest.Broker and have it handled differently only at serialization.